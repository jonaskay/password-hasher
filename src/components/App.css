.App {
  margin: auto;
  max-width: 320px;
}

.App-title {
  margin-bottom: 2rem;
  margin-top: 0;
  text-align: center;
  text-shadow: 2px 4px 3px rgba(0, 0, 0, 0.12);
}

.App-footer {
  border-top: 1px solid rgba(255, 255, 255, 0.12);
  color: rgba(255, 255, 255, 0.7);
  font-size: 0.875rem;
  margin-top: 2rem;
  padding: 1rem 0;
}

.App-footer a {
  color: rgba(255, 255, 255, 0.7);
}

.App-footer a:hover {
  color: rgb(255, 255, 255);
}

.App-footer > span {
  display: inline-block;
  width: 50%;
}
