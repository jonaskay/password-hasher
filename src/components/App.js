import React, { Component } from 'react';
import './App.css';
import PasswordForm from './PasswordForm';
import PasswordResult from './PasswordResult';

class App extends Component {
  render() {
    return (
      <div className="App">
        <h1 className="App-title">Password Hasher</h1>
        <PasswordForm />
        <PasswordResult />
        <footer className="App-footer">
          <span>&copy; 2018 <a href="https://github.com/jonaskay">Jonas Kay</a></span>
          <span style={{textAlign: 'right'}}>
            Based on <a href="https://crypto.stanford.edu/PwdHash/">PwdHash</a>
            </span>
        </footer>
      </div>
    );
  }
}

export default App;
