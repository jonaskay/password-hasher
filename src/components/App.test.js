import React from 'react';
import { shallow, mount } from 'enzyme';
import App from './App';

const mounted = mount(<App />);

describe('<App />', () => {
  it('renders correctly', () => {
    expect(shallow(<App />)).toMatchSnapshot();
  });

  it('handles a domain input change', () => {
    mounted
      .find('input[name="domain"]')
      .simulate('change', { target: { name: 'domain', value: 'foo' } });

    expect(mounted.state().domain).toBe('foo');
  });

  it('handles a password input change', () => {
    mounted
      .find('input[name="password"]')
      .simulate('change', { target: { name: 'password', value: 'foo' } });

    expect(mounted.state().password).toBe('foo');
  });

  it('handles a form submit', () => {
    mounted.find('form').simulate('submit');
    expect(mounted.state().result).toMatchSnapshot();
  });

  it('handles an empty form submit', () => {
    mounted
      .find('input[name="domain"]')
      .simulate('change', { target: { name: 'domain', value: '' } });
    mounted
      .find('input[name="password"]')
      .simulate('change', { target: { name: 'password', value: '' } });
    mounted.find('form').simulate('submit');

    expect(mounted.state().result).toMatchSnapshot();
  });
});

describe('extractDomain', () => {
  it('extracts a domain from URI', () => {
    expect(mounted.instance().extractDomain('http://foobar.example.org')).toBe('example.org');
  });
});

describe('hashPassword', () => {
  it('hashes a password with salt', () => {
    expect(mounted.instance().hashPassword('foo', 'bar')).toMatchSnapshot();
  });
});
