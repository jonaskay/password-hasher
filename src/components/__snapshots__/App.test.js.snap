// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders correctly 1`] = `
<div
  className="App"
>
  <h1
    className="App-title"
  >
    Password Hasher
  </h1>
  <PasswordForm
    onChange={[Function]}
    onSubmit={[Function]}
  />
  <PasswordResult
    result=""
  />
  <footer
    className="App-footer"
  >
    <span>
      Â© 2018 
      <a
        href="https://github.com/jonaskay"
      >
        Jonas Kay
      </a>
    </span>
    <span
      style={
        Object {
          "textAlign": "right",
        }
      }
    >
      Based on 
      <a
        href="https://crypto.stanford.edu/PwdHash/"
      >
        PwdHash
      </a>
    </span>
  </footer>
</div>
`;
